lib/ipmi_fru.c

```diff
@@ -5074,7 +5074,7 @@ ipmi_fru_set_field_string_rebuild(struct ipmi_intf * intf, uint8_t fruId,
 			header.offset.board   += change_size_by_8;
 		}
 		/* Board type field */
-		if ((f_type == 'c' ) || (f_type == 'b' ))
+		if (((f_type == 'c' ) || (f_type == 'b' )) && header.offset.product)
```

Comment:
this line looked wrong before your change, and with your change it looks even more wrong.
the code below is about product section, why at all check for 'b' (board) and 'c' (chassis) type while it must be 'p' (product) instead?